### PR TITLE
[WIP] Test using eth from docker.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by VoR0220** [comment]
You need to create a volume linking the two processes.

**Comment by axic** [comment]
```
Request: {"jsonrpc":"2.0","method":"eth_sendTransaction","params":[{"data":"0x6060604052341561000c57fe5b5b60e28061001b6000396000f30060606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063834ba3d7146044578063b3de648b14606f575bfe5b3415604b57fe5b6051608c565b60405180826000191660001916815260200191505060405180910390f35b3415607657fe5b608a60048080359060200190919050506092565b005b60005481565b80604051808281526020019150506040518091039020600081600019169055505b505600a165627a7a7230582062e72cb9794a085d05970b7e7a3214f0e9d958431f8e3f8fcc9523c2de94f37c0029","from":"0x932238cfa2da73fc69bb57dd450632bba639ecaf","gas":"0x0000000000000000000000000000000000000000000000000000000005f5e100","gasprice":"0x00000000000000000000000000000000000000000000000000005af3107a4000","to":"0x","value":"0x0000000000000000000000000000000000000000000000000000000000000000"}
],"id":6}
Reply: 
/home/travis/build/ethereum/solidity/test/RPCSession.cpp(309): fatal error in "simple_contract": critical check Json::Reader().parse(reply, result, false) failed
```

It seems the RPC response is invalid JSON.

Either our new-line based reader is broken or `eth` sends back an empty response.

**Comment by chriseth** [comment]
The problem is that the eth node somehow shuts down.

**Comment by VoR0220** [comment]
Maybe not enough memory?

**Comment by axic** [comment]
With the latest IPC updates the error is clear, the other side shuts down:

```
Request: {"jsonrpc":"2.0","method":"eth_sendTransaction","params":[{"data":"0x6060604052341561000c57fe5b5b60e28061001b6000396000f30060606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063834ba3d7146044578063b3de648b14606f575bfe5b3415604b57fe5b6051608c565b60405180826000191660001916815260200191505060405180910390f35b3415607657fe5b608a60048080359060200190919050506092565b005b60005481565b80604051808281526020019150506040518091039020600081600019169055505b505600a165627a7a723058206a87c1fa14ce11e72527a63406ca5f9b7cfecd3825cea693602c001929e7d5fc0029","from":"0x446aff6e3a12385b101084d77ccd3bcd9503495a","gas":"0x0000000000000000000000000000000000000000000000000000000005f5e100","gasprice":"0x00000000000000000000000000000000000000000000000000005af3107a4000","to":"0x","value":"0x0000000000000000000000000000000000000000000000000000000000000000"}
],"id":6}
/home/travis/build/ethereum/solidity/test/RPCSession.cpp(116): fatal error in "simple_contract": Reading on IPC failed
```

File: scripts/tests.sh [hunk 1]
```diff
@@ -43,7 +43,8 @@ test "${output//[[:blank:]]/}" = "3"
 if [[ "$OSTYPE" == "darwin"* ]]; then
     ETH_PATH="$REPO_ROOT/eth"
 else
-    ETH_PATH="eth"
```
Comment (axic) [inline scripts/tests.sh]: I think the option should remain to run it locally or via docker. Maybe checking for `$CI`.

Comment (VoR0220) [reply scripts/tests.sh]:
```
If CI || (CanFindDockerBinary && DockerEthImageExists)
     docker eth
else if (canFindEthBinary)
    eth
else
   err
```

Comment (axic) [reply scripts/tests.sh]: @VoR0220 sounds good, do you want to implement it?

Comment (axic) [reply scripts/tests.sh]: I would extend it so we can remove the extra mac check:
```
If CI || (CanFindDockerBinary && DockerEthImageExists)
  docker eth
else if (canFindEthBinary)
  eth
else if (canFindEthBinaryInRepo)
  $REPO_ROOT/eth
else
  err
```

Comment (VoR0220) [reply scripts/tests.sh]: The fact that mac is that way is a problem that we need to solve. It should have been gotten in the install deps script and should be in the path imo.

**Comment by chriseth** [comment]
Using different method now.

**Pull request closed without merging**